### PR TITLE
Closes #1178, #1249: make empty tile container unfocusable

### LINKED ISSUE #1178: Empty tileContainer is focusable
## Steps to reproduce
- Open firefox
- Remove all pins from tileContainer

### Expected behavior
- tileContainer is not focusable

### Actual behavior
- tileContainer is focusable

### PR DESCRIPTION
**Pull request by Baron-Severin**
- Also fixes 1249, which at times led to the unpin icon from being out of sync with the state

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt [hunk 1]
```diff
@@ -406,6 +411,7 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
                 megaTileTryAgainButton.requestFocus()
             }
         }
+        updateOverlayForCurrentState()
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: Think we need to call this for `refreshTilesForInsertion()` as well, for when inserting a pin into empty container

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: Is there any way to pin a tile aside from clicking the icon itself?  The icon is already updated when we do that.

Comment (psymoon) [reply app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: Interesting, so we call `updateOverlyCurrentState()` in `navUrlInput.onFocusChangeListener {}`. So you're right, technically we are updating `pocketVideoMegaTileView.nextFocusDownId` when we navigate down from pin button focus. Seems very fragile though 🤔

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: Oh, I thought you were talking about #1249 , not #1178.  Right, good point.  I'll add it in.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: This is made a little complicated by how many ways there are to add and remove tiles.

Comment (psymoon) [reply app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: Hopefully gets cleaned up with https://github.com/mozilla-mobile/firefox-tv/issues/1085

**Pull request merged**